centml/compiler/backend.py

```diff
@@ -1,44 +1,47 @@
 import os
 import gc
-import pickle
-import hashlib
 import time
-import tempfile
+import hashlib
 import logging
 import weakref
+import tempfile
 import threading as th
 from http import HTTPStatus
 from typing import List, Callable
 import requests
-import torch
 from torch.fx import GraphModule
-import hidet
-from hidet.graph.frontend.torch.interpreter import Interpreter
-from hidet.graph.frontend.torch.dynamo_backends import get_flow_graph
+import torch
 from hidet.runtime.compiled_graph import CompiledGraph
 from centml.compiler.config import config_instance, CompilationStatus
 from centml.compiler.utils import get_backend_compiled_forward_path
 
 
 class Runner:
     def __init__(self, module: GraphModule, inputs: List[torch.Tensor]):
+        if not module:
+            raise Exception("No module provided")
+
         self._module: GraphModule = weakref.ref(module)
         self._inputs: List[torch.Tensor] = inputs
         self.compiled_forward_function: Callable[[torch.Tensor], tuple] = None
         self.lock = th.Lock()
         self.child_thread = th.Thread(target=self.remote_compilation)
 
+        self.serialized_model_dir = tempfile.TemporaryDirectory()  # pylint: disable=consider-using-with
```

Comment:
is it a good practice to leave this directory open? I believe we only need the model_path and input_path. We can create the model_dir inside the _serialize_model_and_inputs() function and set these paths there.

Reply:
Generally it's recommended to use `TemporaryDirectory` as a context manager. That's what the pylint warning I silenced suggests.

Because of this I first tried to use the "lower-level" `tempfile.mkdtemp()`, which unlike `TemporaryDirectory()` will not automatically cleanup the directory and its contents when the directory object is deleted.

Because TemporaryDirectory makes cleaning up much easier I chose to use it but not as a context manager.

Reply:
okay. we can keep it as it is then.